### DISCUSSION TITLE
Recommended way to rebuild live site

### DISCUSSION THREAD

**Discussion by JuanIrache** [vercel/next.js#37218] [Help]
Hello,

I have a Nextjs site and have made some changes to the code. I have it running with **forever**, so I need to rebuild the site, stop the process, and restart it, if I'm not mistaken. So I run something like

> npm run build && npm stop && npm start

(__npm stop__ stops the **forever** process, __npm start__ restarts it with **forever** and __next start__)

The problem is the moment I start __npm run build__ the public site breaks and it is not accessible again until __npm start__. Normally, that's just a few seconds, but also, if something goes wrong with the build, the site breaks until I find a fix.

Is there a way to prebuild the site without affecting the live website and then move the built contents to their folder?

Thank you.

**Comment by GabenGar** [comment]
Where do you deploy it?

**Reply by JuanIrache** [reply]
An Ubuntu VPS

**Reply by GabenGar** [reply]
Are you using the server provided by nextjs or static export?

**Reply by JuanIrache** [reply]
My server runs Apache, I set a ProxyPass rule towards the port used by **next start**. So I guess I'm using the nextjs server, certainly not a static export

**Reply by GabenGar** [reply]
The server provided by nextJS was built with serverless environment in mind. Outside of it you'll have to write your own CI/CD pipeline.
There are two way to go about it: 
1. Rewrite the code for static export. This way you'll only need to build static content and just change the contents of static folder. Obviously not feasible if there is already a lot of code and logic relying on the nextjs server.
2. Build the server, start it and then change the address to listen to on your proxy at runtime. Just make you sure you have the access to the previous build at all times in case the things go south.

**Reply by JuanIrache** [reply]
Thanks. I think something like the second approach might work for mw
